win32/src/host.rs

```diff
@@ -39,14 +39,73 @@ pub trait Window {
 }
 
 #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
-pub enum FileAccess {
-    READ,
-    WRITE,
+#[derive(Debug, Clone)]
+pub struct FileOptions {
+    /// Permit read access.
+    pub read: bool,
+    /// Permit write access.
+    pub write: bool,
+    /// Truncate the file to zero length.
+    pub truncate: bool,
+    /// Create the file if it doesn't exist.
+    pub create: bool,
+    /// Create the file if it doesn't exist, and fail if it does.
```

Comment:
Should this be some kind of enum with create?  Like Option<CreateOpts> which contains fail_if_missing or something?

I don't have any strong feeling here, you have spent more time near this API than me.  In general my thought on this "host" interface is that we should attempt to put any win32-specific weirdness (like this collection of disjoint bools) within the win32 lib and try to put whatever seems like a more sensible interface here, but that is up to your judgement.

Reply:
I’ll ponder a better design of this, but it was meant to map directly to `std::fs::OpenOptions`. So, Rust weirdness, not win32 weirdness 🙂

bitflag could work instead of bools, I guess.

Reply:
Oh mapping to OpenOptions is a superior choice, I missed that.  It lets us fall back on the design choices made by the Rust std lib which are well thought through.  It might even be worth considering just using OpenOptions directly in the future.